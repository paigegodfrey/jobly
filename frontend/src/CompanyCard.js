import React from 'react';
import { Link, BrowserRouter } from 'react-router-dom';
import defaultLogo from "./default-logo.png";
import './Card.css';

const CompanyCard = ({ company = {} }) => {
  const { name, description, logo_url, handle } = company;
  return (
    <BrowserRouter>
      <Link className="CompanyCard Card card" to={`/companies/${handle}`}>
        <div className="card-body">
          <h6 className="card-title d-flex justify-content-between">
            <span className="text-capitalize">{name}</span>
            <img src={logo_url || defaultLogo} alt={`${name} Logo`} />
          </h6>
          <p>{description}</p>
        </div>
      </Link>
    </BrowserRouter>
  );
}

export default CompanyCard;