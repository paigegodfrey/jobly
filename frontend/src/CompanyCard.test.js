import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

it("renders without crashing", function() {
  render(
    <MemoryRouter>
      <CompanyCard />
    </MemoryRouter>
  );
});

it("matches snapshot with logo", function() {
  let company = {
    handle: "rithm",
    name: "Rithm School",
    description: "Become an exceptional developer in 16 weeks.",
    logo_url:
      "https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
  };
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard company={company} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without logo", function() {
  let company = {
    handle: "algo",
    name: "Algo School",
    description: "Become a mediocre developer in 160 weeks."
  };
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard company={company} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});