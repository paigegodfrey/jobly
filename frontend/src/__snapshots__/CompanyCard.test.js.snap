// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard Card card"
    href="/companies/rithm"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title d-flex justify-content-between"
      >
        <span
          class="text-capitalize"
        >
          Rithm School
        </span>
        <img
          alt="Rithm School Logo"
          src="https://pbs.twimg.com/profile_images/770491761412173826/ZUeIa4tw_400x400.jpg"
        />
      </h6>
      <p>
        Become an exceptional developer in 16 weeks.
      </p>
    </div>
  </a>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard Card card"
    href="/companies/algo"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title d-flex justify-content-between"
      >
        <span
          class="text-capitalize"
        >
          Algo School
        </span>
        <img
          alt="Algo School Logo"
          src="default-logo.png"
        />
      </h6>
      <p>
        Become a mediocre developer in 160 weeks.
      </p>
    </div>
  </a>
</DocumentFragment>
`;
